metashare/api/model_mixins.py

```diff
@@ -15,3 +17,24 @@ class Meta:
 
     created_at = models.DateTimeField(auto_now_add=True)
     edited_at = models.DateTimeField(auto_now=True)
+
+
+class PopulateRepoId:
+    def get_repo_id(self, user):
+        """
+        We need to get the repo ID as a particular user, and not all
+        users have access to all repos, so we can't do this in a data
+        migration. Therefore we have an incremental population approach;
+        every time the repo_id needs to be accessed, we use this method,
+        and get it from the model if present, or query the GitHub API as
+        a fallback, assuming that the current user can access the
+        current repo via the repo URL.
+        """
```

Comment:
I don't love this deferring, but understand why it might be preferred. I wonder about (also?) trying to populate `repo_id` on every `save`?